kb-survey/src/app/survey/survey.component.ts

```diff
@@ -0,0 +1,71 @@
+import { Component, OnInit, inject } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import urlConfig from '../config/url.config.json';
+import { ApiBaseService } from '../services/base-api/api-base.service';
+@Component({
+  selector: 'app-survey',
+  templateUrl: './survey.component.html',
+  styleUrls: ['./survey.component.scss'],
+})
+export class SurveyComponent implements OnInit {
+  assessmentResult: any;
+  baseApiService;
+  route;
+  solutionId!: string | null;
+  district: any;
+  state: any;
+  block: any;
+  school: any;
+  role: any;
+  profileDetails:
+    | { district: any; state: any; block: any; school: any; role: any }
+    | undefined;
+  constructor() {
+    this.baseApiService = inject(ApiBaseService);
+    this.route = inject(ActivatedRoute);
+  }
+
+  ngOnInit(): void {
+    this.route.queryParams.subscribe((queryParam) => {
+      this.solutionId = queryParam['solutionId'];
+      this.profileDetails = {
+        district: queryParam['district'],
```

Comment:
In our current context for KB entity type will not be district, block, etc. It might be anything else. How can we make it generic?

Reply:
will have to see, without key name we might not be able to get the data from the route parameters/ query parameters

Reply:
Pushed a change for this code to obtain these values dynamically